src/types.jl

```diff
@@ -13,13 +11,36 @@ For a more concrete definition please refer the following material:
 abstract type AbstractLattice end
 
 unitcellsize(l::AbstractLattice) = 1
-hasunitcell(l::AbstractLattice) = Val{false}()
+hasunitcell(::AbstractLattice) = Val{false}()
 
 _ncoords(::Val{true}, l::AbstractLattice) = 1 + ndims(l)
 _ncoords(::Val{false}, l::AbstractLattice) = ndims(l)
 ncoordinates(l::AbstractLattice) = _ncoords(hasunitcell(l), l)
 
 
+function size(::Val{true}, l::AbstractLattice, dim::Int)
+    if dim > ncoordinates(l)
+        return 1  # copying behaviour of `size` for multidimensional arrays
+    elseif dim == ncoordinates(l)
+        return unitcellsize(l)
+    else
+        return l.dims[dim]
+    end
+end
+size(::Val{false}, l::AbstractLattice, dim::Int) = (dim > ndims(l)) ? 1 : l.dims[dim]
```

Comment:
could you remind me what is the first argument for?

Reply:
whether or not the lattice has a non-trivial unitcell